homeassistant/components/mammotion/config_flow.py

```diff
@@ -0,0 +1,386 @@
+
+    async def async_step_wifi_confirm(
+        self, user_input: dict[str, Any]
+    ) -> ConfigFlowResult:
+        """Confirm device discovery."""
+
+        address = self._config.get(CONF_ADDRESS)
+        name = self._discovered_devices.get(address)
+        mammotion = Mammotion()
+
+        if user_input is not None:
+            account = user_input.get(CONF_ACCOUNTNAME)
+            password = user_input.get(CONF_PASSWORD)
+
+            if self._cloud_client is None:
+                try:
+                    if mammotion.mqtt_list.get(account) is None:
+                        self._cloud_client = await Mammotion().login(account, password)
+                    else:
+                        self._cloud_client = mammotion.mqtt_list.get(
+                            account
+                        ).cloud_client
+                except HTTPException as err:
+                    return self.async_abort(reason=str(err))
+            user_account = (
+                self._cloud_client.mammotion_http.login_info.userInformation.userAccount
+            )
+
+            await self.async_set_unique_id(user_account, raise_on_progress=False)
+            self._abort_if_unique_id_configured()
+
+            return self.async_create_entry(
+                title=user_account,
+                data={
+                    CONF_ACCOUNTNAME: account,
+                    CONF_PASSWORD: password,
+                    CONF_DEVICE_NAME: name,
+                    CONF_USE_WIFI: user_input.get(CONF_USE_WIFI, True),
+                    **self._config,
+                },
+                options={CONF_STAY_CONNECTED_BLUETOOTH: self._stay_connected},
+            )
+        return self.async_abort(reason="missing_wifi_data")
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> OptionsFlow:
+        """Create the options flow."""
+        return MammotionConfigFlowHandler(config_entry)
+
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle reconfiguration."""
+        entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        if TYPE_CHECKING:
+            assert entry
+
+        errors: dict[str, str] | None = None
+        user_input = user_input or {}
+        if user_input:
+            if not errors:
+                return self.async_update_reload_and_abort(
+                    entry,
+                    data={
+                        **entry.data,
+                        **user_input,
+                    },
+                    reason="reconfigure_successful",
+                )
+
+        schema = {
+            vol.Required(
+                CONF_ACCOUNTNAME, default=entry.data.get(CONF_ACCOUNTNAME)
+            ): cv.string,
+            vol.Required(
+                CONF_PASSWORD, default=entry.data.get(CONF_PASSWORD)
+            ): cv.string,
```

Comment:
How many people would not have an account?

Reply:
Also what if they change the password?